opennlp-tools/src/main/java/opennlp/tools/cmdline/langdetect/LanguageDetectorCrossValidatorTool.java

```diff
@@ -0,0 +1,123 @@
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.LinkedList;
+import java.util.List;
+
+import opennlp.tools.cmdline.AbstractCrossValidatorTool;
+import opennlp.tools.cmdline.CmdLineUtil;
+import opennlp.tools.cmdline.TerminateToolException;
+import opennlp.tools.cmdline.params.CVParams;
+import opennlp.tools.cmdline.params.FineGrainedEvaluatorParams;
+import opennlp.tools.langdetect.LanguageDetectorCrossValidator;
+import opennlp.tools.langdetect.LanguageDetectorEvaluationMonitor;
+import opennlp.tools.langdetect.LanguageDetectorFactory;
+import opennlp.tools.langdetect.LanguageDetectorSample;
+import opennlp.tools.util.eval.EvaluationMonitor;
+import opennlp.tools.util.model.ModelUtil;
+
+public final class LanguageDetectorCrossValidatorTool extends
+    AbstractCrossValidatorTool<LanguageDetectorSample,
+        LanguageDetectorCrossValidatorTool.CVToolParams> {
+
+  interface CVToolParams extends CVParams, TrainingParams, FineGrainedEvaluatorParams {
+  }
+
+  public LanguageDetectorCrossValidatorTool() {
+    super(LanguageDetectorSample.class, CVToolParams.class);
+  }
+
+  public String getShortDescription() {
+    return "K-fold cross validator for the learnable Language Detector";
+  }
+
+  public void run(String format, String[] args) {
+    super.run(format, args);
+
+    mlParams = CmdLineUtil.loadTrainingParameters(params.getParams(), false);
+    if (mlParams == null) {
+      mlParams = ModelUtil.createDefaultTrainingParameters();
+    }
+
+    List<EvaluationMonitor<LanguageDetectorSample>> listeners = new LinkedList<>();
+    if (params.getMisclassified()) {
+      listeners.add(new LanguageDetectorEvaluationErrorListener());
+    }
+
+    LanguageDetectorFineGrainedReportListener reportListener = null;
+    File reportFile = params.getReportOutputFile();
+    OutputStream reportOutputStream = null;
+    if (reportFile != null) {
+      CmdLineUtil.checkOutputFile("Report Output File", reportFile);
+      try {
+        reportOutputStream = new FileOutputStream(reportFile);
+        reportListener = new LanguageDetectorFineGrainedReportListener(reportOutputStream);
+        listeners.add(reportListener);
+      } catch (FileNotFoundException e) {
+        throw createTerminationIOException(e);
+      }
+    }
+
+    LanguageDetectorEvaluationMonitor[] listenersArr = listeners
+        .toArray(new LanguageDetectorEvaluationMonitor[listeners.size()]);
+
+    LanguageDetectorCrossValidator validator;
+    try {
+      LanguageDetectorFactory factory = LanguageDetectorFactory.create(params.getFactory());
+      validator = new LanguageDetectorCrossValidator(mlParams,
+          factory, listenersArr);
+
+      validator.evaluate(sampleStream, params.getFolds());
+    } catch (IOException e) {
+      throw new TerminateToolException(-1,
+          "IO error while reading training data or indexing data: " + e.getMessage(), e);
+    } finally {
+      try {
```

Comment:
We should use java resource handling here. I know it is not possible with the current tool base class, should we change it?